arcgisUM.py

```diff
@@ -0,0 +1,206 @@
+# Wrapper around calls to arcgis.  Helps with testing and future changes.
+
+import datetime
+import logging
+
+logger = logging.getLogger(__name__)
+
+from arcgis.gis import GIS
+
+import dateutil.tz
+
+import config
+
+# secrets really is used during (import to change sensitive properties).
+import secrets  # @UnusedImport
+
+import util
+
+##### Improved code tracebacks for exceptions
+import traceback
+
+def handleError(self, record):  # @UnusedVariable
+    traceback.print_stack()
+    
+logging.Handler.handleError = handleError
+#####
+
+TIMEZONE_UTC = dateutil.tz.tzutc()
+RUN_START_TIME = datetime.datetime.now(tz=TIMEZONE_UTC)
+RUN_START_TIME_FORMATTED = RUN_START_TIME.strftime('%Y%m%d%H%M%S')
+
+# Hold parsed options
+options = None
+
+# TODO: required in this module?
+courseLogHandlers = dict()
+courseLoggers = dict()
+
+def getArcGISConnection(securityinfo):
+    """
+    Get a connection object for ArcGIS based on configuration options
+    
+    :return: Connection object for the ArcGIS service
+    :rtype: arcresthelper.orgtools.orgtools
+    :raises: RuntimeError if ArcGIS connection is not valid
+    :raises: arcresthelper.common.ArcRestHelperError for ArcREST-related errors
+    """
+
+    if not isinstance(securityinfo, dict):
+        raise TypeError('Argument securityinfo type should be dict')
+
+    try:
+        arcGIS = GIS(securityinfo['org_url'],
+                     securityinfo['username'],
+                     securityinfo['password']);
+    except RuntimeError as exp:
+        logger.error("RuntimeError: getArcGISConnection: {}".format(exp))
+        raise RuntimeError(str('ArcGIS connection invalid: {}'.format(exp)))
+    
+    return arcGIS
+
+def getArcGISGroupByTitle(arcGISAdmin, title):
+    """
+    Given a possible title of a group, search for it in ArcGIS
+    and return a Group object if found or None otherwise.
+
+    :param arcGISAdmin: ArcGIS Administration REST service connection object
+    :type arcGISAdmin: arcrest.manageorg.administration.Administration
+    :param title: Group title to be found
+    :type title: str
+    :return: ArcGIS Group object or None
+    :rtype: Group or None
+    """
+    searchString = "title:"+title
+    logger.debug("group search string: original: {}".format(searchString))
+
+    # quote characters that are special in the group search.
+    searchString = searchString.translate(str.maketrans({"?":  r"\?","*":  r"\*"}))
+
+    logger.debug("group search string: escaped: {}".format(searchString))
+    
+    try:
+        gis_groups = arcGISAdmin.groups.search(searchString)
+    except RuntimeError as exp:
+        logger.error("arcGIS error finding group: {} exception: {}".format(searchString,exp))
+    
+    if len(gis_groups) > 0:
```

Comment:
what will the value be for `gis_groups` in case of `RuntimeError`?  I suppose it will be `None`  ,then `len(gis_groups)` doesn't this will throw an error?

Reply:
It will now return None from the exception.